From 0883d2af0fc815942f20c991ba342e6d76aba6bc Mon Sep 17 00:00:00 2001
From: Patrick Boog <jonesthefox@icloud.com>
Date: Fri, 30 Aug 2024 04:53:38 +0200
Subject: [PATCH] Fix ELF magic number does not match

U-boot build fails with:

---
  LD      spl/u-boot-spl
  OBJCOPY spl/u-boot-spl-nodtb.bin
  SYM     spl/u-boot-spl.sym
  CAT     spl/u-boot-spl-dtb.bin
  COPY    spl/u-boot-spl.bin
  BINMAN  .binman_stamp
Wrote map file './simple-bin.map' to show errors
binman: Node '/binman/simple-bin/fit': subnode 'images/@atf-SEQ': Failed to read ELF file: Magic number does not match
make: *** [Makefile:1126: .binman_stamp] Fehler 1
---

Here is a small patch as tobetter mentioned in [1].

[1] https://forum.odroid.com/viewtopic.php?p=385434&sid=60b09f610b633db773bda476b96f8e6a#p385434

---
 arch/arm/dts/rockchip-u-boot.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/dts/rockchip-u-boot.dtsi b/arch/arm/dts/rockchip-u-boot.dtsi
index c8c928c7e50..2f36076c5bd 100644
--- a/arch/arm/dts/rockchip-u-boot.dtsi
+++ b/arch/arm/dts/rockchip-u-boot.dtsi
@@ -80,6 +80,7 @@
 				};
 
 #ifdef CONFIG_ARM64
+#if !defined(CONFIG_TARGET_ODROID_M1S_RK3566)
 				@atf-SEQ {
 					fit,operation = "split-elf";
 					description = "ARM Trusted Firmware";
@@ -99,6 +100,7 @@
 					};
 #endif
 				};
+#endif
 				@tee-SEQ {
 					fit,operation = "split-elf";
 					description = "TEE";
-- 
2.39.2

